Toggle table of contents

GDSNewt opened the pull request:
This change updates the backend to expose the hide_contents_list property to frontend applications.

The property is now defined at the schema level, providing more explicit control over its rendering. We recently implemented a [similar feature for the metadata block ](https://github.com/alphagov/specialist-publisher/pull/3076). 

Tests in this PR depend on example schemas from the Publishing API repository, which may cause CI failures until the related changes are merged.

**Associated PRs**

[Publishing API](https://github.com/alphagov/publishing-api/pull/3265)
[Frontend](https://github.com/alphagov/frontend/pull/4740)
[Government-Frontend](https://github.com/alphagov/government-frontend/pull/3645)


[Trello](https://trello.com/c/2eakEGOo/3570-make-it-possible-to-hide-the-table-of-contents-on-specialist-documents)

GDSNewt committed:
Surface show_table_of_contents property

GDSNewt committed:
Add show_table_of_contents option to new/edit finder admin page

lauraghiorghisor-tw reviewed (approved):
LGTM - but could rename to `show_` as Leena suggested elsewhere. (Think the test failure is expected and should go green once pub api is merged.)

GDSNewt committed:
Add show_table_of_contents value to each finder schema

We can assume that most finders should have this value set as true.

lauraghiorghisor-tw committed:
Set show_table_of_contents to false for HMRC

The reason why the field was added to the config in the first place was to be able to turn it off for the HMRC finder.
